pandas/tests/reshape/test_reshape.py

```diff
@@ -991,3 +991,54 @@ def test_non_unique_idvars(self):
         })
         with pytest.raises(ValueError):
             wide_to_long(df, ['A_A', 'B_B'], i='x', j='colname')
+
+    def test_cast_j_int(self):
```

Comment:
can you add an example where the cast fails? e.g. columns are ['A_1', 'A_foo']...